platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java

```diff
@@ -217,7 +218,7 @@ public static MapboxMapOptions createFromAttributes(@NonNull Context context, @N
       Drawable compassDrawable = typedArray.getDrawable(
         R.styleable.mapbox_MapView_mapbox_uiCompassDrawable);
       if (compassDrawable == null) {
-        compassDrawable = ContextCompat.getDrawable(context, R.drawable.mapbox_compass_icon);
+        compassDrawable = ResourcesCompat.getDrawable(context.getResources(), R.drawable.mapbox_compass_icon, null);
```

Comment:
AFAIK ContextCompat isn't deprecated and will not be deprecated. It provides compatibility with LOLLIPOP in `ContextCompatApi21.java`. Feel free to link more information on this.